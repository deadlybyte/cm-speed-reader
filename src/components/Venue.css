.CM-venue-block {
  background-color: #848284;
  border: 3px outset;
  border-top-color: #ded7d6;
  border-left-color: #ded7d6;
  border-bottom-color: #5a5552;
  border-right-color: #5a5552;
  padding: 2px;
  text-align: center;
  min-height: 40px;
  margin: 10px 0;
}

.CM-venue-block span {
  color: white;
  font-size: x-large;
}

.CM-venue-block p {
  margin: 12px;
  word-wrap: break-word;
}
