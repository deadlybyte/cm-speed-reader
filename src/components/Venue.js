// @flow
import React from 'react';
import './Venue.css';

const Venue = props => {
  const { name, location } = props;
  return (
    <div className="CM-venue-block">
      <p>
        <span>
          {name ? name : 'Unknown'}, {location ? location : 'Unknown'}
        </span>
      </p>
    </div>
  );
};

export default Venue;
