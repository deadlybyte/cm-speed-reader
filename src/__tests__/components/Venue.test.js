import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Venue from '../../components/Venue';

describe('Venue', () => {
  describe('render', () => {
    it('should render without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<Venue />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    it('should render as Unknown venue', () => {
      const componentUnderTest = renderer.create(<Venue />)
        .toJSON();
      expect(componentUnderTest).toMatchSnapshot();
    });

    it('should render venue from props', () => {
      const componentUnderTest = renderer.create(<Venue name="Molineux Stadium" location="Wolverhampton" />)
        .toJSON();
      expect(componentUnderTest).toMatchSnapshot();
    });

    it('should render venue name when only name set and default to Unknown location', () => {
      const componentUnderTest = renderer.create(<Venue name="Molineux Stadium" />)
        .toJSON();
      expect(componentUnderTest).toMatchSnapshot();
    });

    it('should render venue location when only location set and default to Unknown name', () => {
      const componentUnderTest = renderer.create(<Venue location="Wolverhampton" />)
        .toJSON();
      expect(componentUnderTest).toMatchSnapshot();
    });
  });
});
