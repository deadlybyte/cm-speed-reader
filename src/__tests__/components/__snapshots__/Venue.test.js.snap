// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Venue render should render as Unknown venue 1`] = `
<div
  className="CM-venue-block"
>
  <p>
    <span>
      Unknown
      , 
      Unknown
    </span>
  </p>
</div>
`;

exports[`Venue render should render venue from props 1`] = `
<div
  className="CM-venue-block"
>
  <p>
    <span>
      Molineux Stadium
      , 
      Wolverhampton
    </span>
  </p>
</div>
`;

exports[`Venue render should render venue location when only location set and default to Unknown name 1`] = `
<div
  className="CM-venue-block"
>
  <p>
    <span>
      Unknown
      , 
      Wolverhampton
    </span>
  </p>
</div>
`;

exports[`Venue render should render venue name when only name set and default to Unknown location 1`] = `
<div
  className="CM-venue-block"
>
  <p>
    <span>
      Molineux Stadium
      , 
      Unknown
    </span>
  </p>
</div>
`;
